src/plugins/kv-analytics-plugin.js

```diff
@@ -10,28 +10,38 @@ export default Vue => {
 			snowplowLoaded = inBrowser && typeof window.snowplow === 'function';
 		},
 		pageview: (to, from) => {
-			const toUrl = window.location.origin + to.path;
-			const asyncFromUrl = window.location.origin + from.path;
+			if (!inBrowser) return false;
+
+			let toUrl = window.location.href;
+			let fromUrl = document.referrer;
+			// update urls for async page changes
+			if (to.matched && to.matched.length && from.matched && from.matched.length) {
+				toUrl = window.location.origin + to.fullPath;
+				fromUrl = window.location.origin + from.fullPath;
+			}
```

Comment:
Update our to and from urls if we are going between Vue pages that have Vue Router information.